crates/ty_python_semantic/src/types/context.rs

```diff
@@ -97,7 +97,9 @@ impl<'db, 'ast> InferContext<'db, 'ast> {
     }
 
     pub(crate) fn extend(&mut self, other: &TypeCheckDiagnostics) {
-        self.diagnostics.get_mut().extend(other);
+        if !self.is_in_multi_inference() {
```

Comment:
Should it be the other way where we don't extend self if other is a multi inference context?

Reply:
Okay, I think I figured it out. `other` is guaranteed to be empty if it is a `multi_inference` region. That's why we only have to check self

Reply:
Right, the issue is that if `other` was created using a different context (e.g. because it was executed as a salsa query), we still want to respect to context of `self`.